### PR TITLE
Promote Service Controller e2e test to Conformance

### LINKED ISSUE #78835: Promote ServiceAccount Tests to Conformance
Using the list of tested stable endpoints referenced in https://github.com/kubernetes/kubernetes/issues/78748, we can pick out the endpoints which are part of the ServiceAccount kind:

[replaceCoreV1NamespacedServiceAccount](https://apisnoop.cncf.io/?zoomed=operationId-stable-core-replaceCoreV1NamespacedServiceAccount&showUntested=false&showConformanceTested=false)

This is a stable core endpoint which has existing testing, potentially making for low-hanging fruit to upgrade to Conformance.

/area test
/area conformance
/sig testing

### PR DESCRIPTION
**Pull request by devaips**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Promotes the following E2E test to Conformance:
 `test/e2e/auth/service_accounts.go: "should ensure a single API token exists"`

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/78835

**Special notes for your reviewer**:

Part of Umbrella Issue https://github.com/kubernetes/kubernetes/issues/78748

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/area conformance
/area test

@kubernetes/sig-architecture-pr-reviews 
@kubernetes/cncf-conformance-wg
@kubernetes/sig-auth-pr-reviews

### CODE REVIEW

File: test/e2e/auth/service_accounts.go [hunk 1]
```diff
@@ -46,7 +46,15 @@ var inClusterClientImage = imageutils.GetE2EImage(imageutils.InClusterClient)
...
-	ginkgo.It("should ensure a single API token exists", func() {
+
+	/*
+	   Release: v1.16
+	   Testname: Service Account: Single Secret Reference
+	   Description: Ensure that a given Service Account has a secret reference
+                    There MUST only be one secret reference. The reference MUST
+                    be able to be deleted, and a new reference added.
+	*/
+	framework.ConformanceIt("should ensure a single API token exists", func() {
```
Comment (draveness) [inline test/e2e/auth/service_accounts.go]: /ok-to-test
/priority important-soon

**Comment by smarterclayton** [comment]
/hold

This test is not appropriate to promote to conformance, because it prevents conformant implementations that register additional service account tokens from being successful.

@kubernetes/sig-auth-api-reviews, I think we need to discuss whether this e2e test is appropriate, or whether it should be less restrictive.

**Comment by liggitt** [comment]
agree this is not a good candidate for conformance. the goal of the test originally was to catch flutter in the service account controller code.

**Comment by liggitt** [comment]
closing this PR (since single token is not conformant behavior we care about)
/close

can discuss ways to better target what the test is actually checking separately

**Pull request closed without merging**